flooddrake/timestepper.py

```diff
@@ -211,53 +215,49 @@ def stepper(self, t_start, t_end, w):
         bout = Function(self.v_h).project(self.b_)
         bout_file = File("b.pvd")
 
-        hout_file.write(hout.project(h + self.b_))
+        hout_file.write(hout.project(self.h + self.b_))
         bout_file.write(bout)
 
         self.t = 0
 
         for i in range(Nt):
 
-            w_ = self.__solver_variable(w)
+            solve(self.L == 0, self.w_, nest=False,
+                  solver_parameters={'ksp_type': 'preonly',
+                                     'pc_type': 'lu'})
```

Comment:
This is a nonlinear solve.  Is that actually what you wanted?  I also recommend that you pull the solver out of the time loop:

```
problem = NonlinearVariationalProblem(self.L, self._w)
solver = NonlinearVariationalSolver(problem, ...)

for i in range(Nt):
    solver.solve()
    self.w.assign(self.w_)
    self.__update_slope_modification()
    solver.solver()
    ...
```